### ISSUE TITLE
[Q] hyperparameter sweeps

### ISSUE THREAD

**Issue by ryanliu30** [wandb/wandb#5980]
Hi, I am using wandb to run a hyperparameter sweep. I noticed that in the parallel coordinates and parameter importance plots they automatically default to the metric reported in the last epoch. I wonder is that the intended behavior? In my opinion, taking the maximum/minimum throughout the run would make more sense.

**Comment by sydholl** [comment]
Internal GitHub User commented: 
Hi, I am using wandb to run a hyperparameter sweep. I noticed that in the parallel coordinates and parameter importance plots they automatically default to the metric reported in the last epoch. I wonder is that the intended behavior? In my opinion, taking the maximum/minimum throughout the run would make more sense.

**Comment by rsanandres-wandb** [comment]
Hello @ryanliu30 !

Could you clarify what you mean by "they automatically default to the metric reported in the last epoch"? Which metric automatically defaults in the parallel coordinates and parameter importance plots. 

In general, the common usecase for parallel coordinates is to have the config values as the conditions and a value as the result (such as Validation Loss). Are you asking for taking the maximum/minimum of this value as part of the parallel coordinates/parameter importance plot?

**Comment by ryanliu30** [comment]
Hi @rsanandres-wandb. For example, I am logging validation AUC to wandb to monitor the performance of my model. It might look like: 
![W B Chart 8_1_2023, 3_11_28 PM](https://github.com/wandb/wandb/assets/72370449/e6c91fe5-9de3-454e-87d4-df0647342848)
Where we can see AUC is fluctuating. In importance and parallel coordinates plots they take the last epoch's value, which is not a fair comparison for those runs experiencing a drop during the last epoch. While testing, we take the best-performing epoch's checkpoint in validation to avoid overfitting issues, but the hyperparameter sweep does not take that into account. I would suggest adding an optional configuration allowing the user to specify how to determine which epoch to use (e.g. "max", "min", "avg", "last", etc.)

**Comment by rsanandres-wandb** [comment]
Ah okay, that makes sense. I can certainly write a feature request for our team to look at this but your explanation is understandable.

**Comment by ryanliu30** [comment]
Thank you @rsanandres-wandb please make the feature request.

**Comment by sydholl** [comment]
WandB Internal User commented: 
ryanliu30 commented: 
Hi @rsanandres-wandb. For example, I am logging validation AUC to wandb to monitor the performance of my model. It might look like: 
![W B Chart 8_1_2023, 3_11_28 PM](https://github.com/wandb/wandb/assets/72370449/e6c91fe5-9de3-454e-87d4-df0647342848)
Where we can see AUC is fluctuating. In importance and parallel coordinates plots they take the last epoch's value, which is not a fair comparison for those runs experiencing a drop during the last epoch. While testing, we take the best-performing epoch's checkpoint in validation to avoid overfitting issues, but the hyperparameter sweep does not take that into account. I would suggest adding an optional configuration allowing the user to specify how to determine which epoch to use (e.g. "max", "min", "avg", "last", etc.)

**Comment by sydholl** [comment]
WandB Internal User commented: 
ryanliu30 commented: 
Thank you @rsanandres-wandb please make the feature request.
